src/internal/pjsdb/job_crud.go

```diff
@@ -307,25 +308,80 @@ func DeleteJob(ctx context.Context, tx *pachsql.Tx, id JobID) ([]JobID, error) {
 // done timestamp in database.
 func ErrorJob(ctx context.Context, tx *pachsql.Tx, jobID JobID, errCode pjs.JobErrorCode) error {
 	ctx = pctx.Child(ctx, "complete error")
+	errStr := errEnumToString(errCode)
 	_, err := tx.ExecContext(ctx, `
 		UPDATE pjs.jobs
 		SET done = CURRENT_TIMESTAMP, error = $1
-		WHERE id = $2 AND error IS NULL
-	`, errCode, jobID)
+		WHERE id = $2 AND error IS NULL AND done IS NULL
+	`, errStr, jobID)
 	if err != nil {
 		return errors.Wrapf(err, "error job: update error and state to done")
 	}
 	return nil
 }
 
+// ErrEnumToString fills in the gap between pjs proto error enum and database error enum
+func errEnumToString(errCode pjs.JobErrorCode) string {
+	converter := map[pjs.JobErrorCode]string{
+		pjs.JobErrorCode_DISCONNECTED: "disconnected",
+		pjs.JobErrorCode_FAILED:       "failed",
+		pjs.JobErrorCode_CANCELED:     "cancelled",
+	}
+	return converter[errCode]
+}
+
+// StringToErrEnum fills in the gap between database error enum and pjs proto error enum
+func stringToErrEnum(errStr string) pjs.JobErrorCode {
+	converter := map[string]pjs.JobErrorCode{
+		"disconnected": pjs.JobErrorCode_DISCONNECTED,
+		"failed":       pjs.JobErrorCode_FAILED,
+		"cancelled":    pjs.JobErrorCode_CANCELED,
+	}
+	return converter[errStr]
+}
+
+func ToJobInfo(job Job) (*pjs.JobInfo, error) {
```

Comment:
In a follow-up story this could be private (i.e. `toJobInfo`), `pjsdb.Job` could be private too and functions which currently return `pjsdb.Job`s could return `*pjs.JobInfo`s instead.